kolibri/core/tasks/job.py

```diff
@@ -260,11 +266,32 @@ def update_progress(self, progress, total_progress):
             return
         self.progress = progress
         self.total_progress = total_progress
-        self.storage.update_job_progress(self.job_id, progress, total_progress)
+        last_saved_progress_float = self._float_progress(
+            self._last_saved_progress, self._last_saved_total_progress
+        )
+        if isinstance(extra_metadata, dict):
+            self._update_meta(**extra_metadata)
+        if (
+            # Total progress has been changed.
+            (self.total_progress != self._last_saved_total_progress)
+            # Progress is now 'complete' for the first time.
+            or (last_saved_progress_float < 1 and self.percentage_progress >= 1)
```

Comment:
I appreciate the specificity of the local variables, although a nitpick that comparing them with `self.precentage_progress` does not have the same specificity. Not a big deal as you can quickly jump to the computation for that property.